### PR TITLE
CrossmintAuthProvider: move from auth-core to react-ui

### PR DESCRIPTION
**Pull request by jmderby**
## Description

We're relocating the `CrossmintAuthProvider` from `auth-core` to `react-ui` to better organize our React-specific components. This move not only improves code structure and maintainability but also facilitates our upcoming T3 changes, specifically the migration of our auth flow from the main repo to our SDK. Additionally, this change orchestrates easier implementation of the passkeys UI within this package, streamlining our authentication-related components.

## Test plan

regular auth flow still works

## Package updates

**@crossmint/client-sdk-auth-core** 
- removed CrossmintAuthProvider and anything relating to react

**@crossmint/client-sdk-react-ui**
- added CrossmintAuthProvider

**@crossmint/common-sdk-base**
- added helper fn for api key validation and getting base url

**@crossmint/client-sdk-base**
- removed helper fn for api key validation and getting base url

### CODE REVIEW

File: packages/client/ui/react-ui/src/services/logger.ts
[the remark this answers is not in the document]
Comment (jmderby) [reply packages/client/ui/react-ui/src/services/logger.ts:1]: done

File: packages/client/ui/react-ui/src/utils/constants.ts
[the remark this answers is not in the document]
Comment (jmderby) [reply packages/client/ui/react-ui/src/utils/constants.ts:1]: We still use `SESSION_PREFIX` so kept that but removed other

File: packages/client/ui/react-ui/src/components/index.ts [hunk 1]
```diff
@@ -3,3 +3,4 @@ export * from "./CrossmintNFTDetail";
 
 export * from "./embed";
 export * from "./hosted";
+export * from "./auth";
```
[the remark this answers is not in the document]
Comment (jmderby) [reply packages/client/ui/react-ui/src/components/index.ts:6]: right, removed...

File: packages/client/auth-core/src/components/ActionModal.tsx
[the remark this answers is not in the document]
Comment (jmderby) [reply packages/client/auth-core/src/components/ActionModal.tsx:1]: ActionModal and Authmodal all live within new AuthModal component in react-ui

File: packages/client/ui/react-ui/src/services/CrossmintAuthService.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { APIErrorService, BaseCrossmintService } from "@crossmint/client-sdk-base";
+
+import { authLogger } from "./logger";
+
+export class CrossmintAuthService extends BaseCrossmintService {
```
[the remark this answers is not in the document]
Comment (jmderby) [reply packages/client/ui/react-ui/src/services/CrossmintAuthService.ts]: good call, moved into helper fn which the base crossmint service uses as well

File: packages/client/base/src/services/api/helpers/validateApiKeyAndGetCrossmintBaseUrl.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+import { CROSSMINT_DEV_URL, CROSSMINT_PROD_URL, CROSSMINT_STG_URL } from "../../../consts";
+
+const urlMap: Record<APIKeyEnvironmentPrefix, string> = {
+    development: CROSSMINT_DEV_URL,
+    staging: CROSSMINT_STG_URL,
+    production: CROSSMINT_PROD_URL,
+};
+
+export function validateApiKeyAndGetCrossmintBaseUrl(apiKey: string) {
```
Comment (devlyn37) [inline packages/client/base/src/services/api/helpers/validateApiKeyAndGetCrossmintBaseUrl.ts:11]: Should this go in the base package instead, where the other API key validation related code lives. Server SDKs probably want this too right?

**Review by devlyn37 — APPROVED** [review]
Nice! Left a couple comments but LGTM

File: packages/client/base/src/services/api/helpers/validateApiKeyAndGetCrossmintBaseUrl.ts (hunk 3 above)
Comment (jmderby) [reply packages/client/base/src/services/api/helpers/validateApiKeyAndGetCrossmintBaseUrl.ts]: done

File: packages/client/ui/react-ui/src/providers/CrossmintAuthProvider.tsx [hunk 4]
```diff
@@ -12,12 +15,64 @@ type CrossmintAuthProviderProps = {
...
 
+export const AuthContext = createContext({
+    login: () => {},
+    logout: () => {},
+    jwt: undefined as string | undefined,
+});
+
 export function CrossmintAuthProvider({ embeddedWallets, children, appearance }: CrossmintAuthProviderProps) {
     const { crossmint, setJwt } = useCrossmint("CrossmintAuthProvider must be used within CrossmintProvider");
+    const crossmintBaseUrl = validateApiKeyAndGetCrossmintBaseUrl(crossmint.apiKey);
```
[the remark this answers is not in the document]
Comment (jmderby) [reply packages/client/ui/react-ui/src/providers/CrossmintAuthProvider.tsx:25]: i think its negligible

**Pull request merged**